infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/Orchestrator.java

```diff
@@ -0,0 +1,89 @@
+package org.corfudb.infrastructure.orchestrator;
+
+import io.netty.channel.ChannelHandlerContext;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.infrastructure.IServerRouter;
+import org.corfudb.infrastructure.orchestrator.workflows.AddNode;
+import org.corfudb.protocols.wireprotocol.CorfuMsgType;
+import org.corfudb.protocols.wireprotocol.CorfuPayloadMsg;
+import org.corfudb.protocols.wireprotocol.orchestrator.AddNodeRequest;
+import org.corfudb.protocols.wireprotocol.orchestrator.OrchestratorRequest;
+import org.corfudb.runtime.CorfuRuntime;
+
+import javax.annotation.Nonnull;
+import java.util.UUID;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * The orchestrator is a stateless service that runs on all management servers and its purpose
+ * is to execute workflows. A workflow defines multiple smaller actions that run on the cluster
+ * in order to achieve a bigger goal. For example, growing the cluster. Initiated through RPC,
+ * the orchestrator will create a workflow instance and attempt to execute all its actions.
+ *
+ * <p>
+ * Created by Maithem on 10/25/17.
+ */
+
+@Slf4j
+public class Orchestrator {
+
+    final Callable<CorfuRuntime> runtime;
+
+    public Orchestrator(@Nonnull Callable<CorfuRuntime> runtime) {
+        this.runtime = runtime;
+    }
+
+    public void handle(@Nonnull CorfuPayloadMsg<OrchestratorRequest> msg,
+                       @Nonnull ChannelHandlerContext ctx,
+                       @Nonnull IServerRouter r) {
+
+        OrchestratorRequest req = msg.getPayload();
+        dispatch(req);
+        r.sendResponse(ctx, msg, CorfuMsgType.ORCHESTRATOR_RESPONSE.msg());
+    }
+
+    void dispatch(@Nonnull OrchestratorRequest req) {
+        CompletableFuture.runAsync(() -> {
+            run(getWorkflow(req));
+        });
+    }
+
+    /**
+     * Create a workflow instance from an orchestrator request
+     * @param req Orchestrator request
+     * @return Workflow instance
+     */
+    @Nonnull
+    private Workflow getWorkflow(@Nonnull OrchestratorRequest req) {
+        switch (req.getWorkFlowType()) {
+            case ADD_NODE:
+                AddNodeRequest addNodeRequest = (AddNodeRequest) req;
+                return new AddNode(addNodeRequest.getEndpoint());
+            default:
+                throw new RuntimeException("Unknown workflow");
+        }
+    }
+
+    /**
+     * Run a particular workflow, which entails executing all its defined
+     * actions
```

Comment:
Be more specific. Do all the actions run in order? Put it in the javadoc.

Reply:
Done.

Reply:
I'll document this in the java doc. its superfluous to document this every time "workflow actions" is mentioned.